mllp_http/http2mllp.py

```diff
@@ -68,57 +69,42 @@ def send(self, data):
         if connection is None:
             connection = self._connect()
         response = connection.send(data)
-        if self.options.max_messages <= connection.message_count:
+        if self.options.max_messages > 0 and self.options.max_messages <= connection.message_count:
             connection.close()
         else:
             connection.last_update = time.monotonic()
             with self.lock:
                 self.connections.append(connection)
         return response
 
-
 class MllpConnection:
     def __init__(self, socket):
-        self.closed = False
-        self.last_update = None
-        self.message_count = 0
-        self.socket = socket
-
-    def close():
-        self.close = True
-        self.socket.shutdown()
-        self.socket.close()
-
-
-class MllpConnection:
-    def __init__(self, socket):
-        self.address = address
         self.cancel = None
         self.closed = False
-        self.responses = read_mllp(read_socket_bytes(self.socket))
         self.socket = socket
+        self.responses = read_mllp(read_real_socket_bytes(self.socket))
+        self.message_count=0
+        self.last_update = time.monotonic()
 
-    def close():
+    def close(self):
         self.closed = True
         self.socket.close()
 
-    def send(data):
-        write_mllp(self.socket, data)
-        self.socket.flush()
+    def send(self,data):
+        write_mllp_socket(self.socket, data)
+        # self.socket.flush()
```

Comment:
What is the purpose of this change?

Reply:
currently, http2mllp uses socket vs stream that mllp2http uses. so it require slightly different read/write function to send/receive